Fatma Homework week2, React

FatmaHameed opened the pull request:
The project is deployed in this link: [hackyourweather](https://hackyourweatherweek2.netlify.app/)

nick-barth commented on week2/Project/hackyourweather/src/App.js:
```diff
@@ -0,0 +1,19 @@
+import Title from './components/header'
+import './App.css';
+import Main from './components/Main';
+// import InputCity from './components/input'
```
remove

FatmaHameed replied:
OK, Done

nick-barth commented on week2/Project/hackyourweather/src/App.js:
```diff
@@ -0,0 +1,19 @@
+    <div className="App">
+      <header className="header">
+      <Title title ="Weather" />
+      {/* <WeatherBox /> */}
```
:(

FatmaHameed replied:
Done. It is removed

nick-barth commented on week2/Project/hackyourweather/src/App.js:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <div className="App">
+      <header className="header">
+      <Title title ="Weather" />
```
Please install a code formatter so we do not have formatting mistakes.

FatmaHameed replied:
Done. I used it for formatting code now

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+import React, {useState}  from 'react';
+import WeatherBox from './weatherBox';
+import Spinner from './Spinner';
+import ErrorMessage from './errorText'
```
components should be class cased, ErrorText, WeatherBox, Spinner etc.

FatmaHameed replied:
Done. I changed the components files names

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+   const [isLoading, setLoading] = useState(false);
+  const [isError, setError] = useState({});
+  const API_KEY = process.env.REACT_APP_OPENWEATHERMAP_API_KEY
+  // console.log(API_KEY);
```
remove

FatmaHameed replied:
Done

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+  const [cityName, setCityName] = useState('');
+   const [isLoading, setLoading] = useState(false);
+  const [isError, setError] = useState({});
+  const API_KEY = process.env.REACT_APP_OPENWEATHERMAP_API_KEY
```
pull outside of component.  good use of uppercase and underscore here :)

FatmaHameed replied:
OK.

FatmaHameed replied:
Thank you

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+  const [isError, setError] = useState({});
+  const API_KEY = process.env.REACT_APP_OPENWEATHERMAP_API_KEY
+  // console.log(API_KEY);
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${API_KEY}&units=metric`
```
also pull this out of the component.  you do not need to redeclare this every time a component rerenders.

FatmaHameed replied:
This which I could not put out of the component because it is using cityName variable which cannot be declared outside

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+      const data = await response.json();
+        if (data.cod < 400) {
+        setCity(data);
+        console.log(data)
```
ILLEGAL police police

FatmaHameed replied:
Ok. It is removed

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+      }
+     await setCityName('');
+    } catch (error) {
+    console.log(error);
```
no

FatmaHameed replied:
Ok

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+     await setCityName('');
+    } catch (error) {
+    console.log(error);
+    // console.log(err);
```
extra no

FatmaHameed replied:
OK

nick-barth commented on week2/Project/hackyourweather/src/components/Main.js:
```diff
@@ -0,0 +1,61 @@
+}
+export default Main; 
+
+
```
5 extra trailing lines.

nick-barth reviewed (changes requested):
Please install prettier and have it format this code for you, it is difficult to read without it.  You will notice the difference as soon as you try it.  In addition, console.log should never make it to github.

FatmaHameed commented:
@nick-barth , Thank you for reviewing my code and for your comments. I removed the commented codes and console logs.  I also used prettier for code formatting and addressed other mentioned comments. I pushed my work again.

Thanks again for your feedback.

wouterkleijn commented:
Great job @FatmaHameed
